tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +91,494 @@ mvn package
 
-Our `profile` document is ready to be persisted to the database.  We create call to the `collection` using the local variable `profileCol` and then call the `insert` method  and passing it the UUID from the Profile's `getPid` method as the key. Once the document is inserted we then return the document saved and the result all as part of the same object back to the user.
+An example of the pattern for AirlineRepository is shown below. For the full code, see the [AirlineRepositoryImpl.java](https://github.com/couchbase-examples/java-springboot-quickstart/blob/main/src/main/java/org/couchbase/quickstart/springboot/repositories/AirlineRepositoryImpl.java).
 
 ```java
-Profile profile = userProfile.getProfile();
-profileCol.insert(profile.getPid(), profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
-```
+public interface AirlineRepository {
+    Airline findById(String id);
 
-> *from controllers/ProfileController.java*
+    Airline save(Airline airline);
 
-## GET a Profile by Key
+    Airline update(String id, Airline airline);
 
-Navigate to the getProfile method in the ProfileController file in the controllers folder.  We only need the profile ID `pid` or our `Key` from the user to retrieve a particular profile document using a basic key-value operation which is passed in the method signature as a string.  Since we created the document with a unique key we can use that key to find the document in the scope and collection it is stored in.
+    void delete(String id);
+
+    List<Airline> findAll(int limit, int offset);
+
+    List<Airline> findByCountry(String country, int limit, int offset);
+
+    List<Airline> findByDestinationAirport(String destinationAirport, int limit, int offset);
+}
+
+@Repository
+public class AirlineRepositoryImpl implements AirlineRepository {
+
+    private final Cluster cluster;
+    private final Collection airlineCol;
+    private final CouchbaseConfig couchbaseConfig;
+
+    public AirlineRepositoryImpl(Cluster cluster, Bucket bucket, CouchbaseConfig couchbaseConfig) {
+        this.cluster = cluster;
+        this.airlineCol = bucket.scope("inventory").collection("airline");
+        this.couchbaseConfig = couchbaseConfig;
+    }
+
+    @Override
+    public Airline findById(String id) {
+        return airlineCol.get(id).contentAs(Airline.class);
+    }
+
+    @Override
+    public Airline save(Airline airline) {
+        airlineCol.insert(airline.getId(), airline);
+        return airline;
+    }
+
+    @Override
+    public Airline update(String id, Airline airline) {
+        airlineCol.replace(id, airline);
+        return airline;
+    }
+
+    @Override
+    public void delete(String id) {
+        airlineCol.remove(id);
+    }
+
+}
 
-```java
-Profile profile = profileCol.get(pid).contentAs(Profile.class);
-return ResponseEntity.status(HttpStatus.OK).body(profile);
 ```
 
-> *from getProfile method of controllers/ProfileController.java*
+## Mapping Workflow
```

Comment:
What is Mapping Workflow? You need to mention what that is before going deep down into it.

Reply:
okay added an explaination for it